azure-pipelines.yml

```diff
@@ -89,4 +89,34 @@ jobs:
             SERVICE: 'Anonymizer'
             WORKING_FOLDER: 'presidio-anonymizer'
 
-        
\ No newline at end of file
+    - job: FunctionalTests
+      dependsOn: 'SecurityAnalysis'
+      pool:
+        vmImage: 'ubuntu-16.04'
+
+      steps:
+        - task: DockerCompose@0
+          displayName: Start Presidio Cluster
+          inputs:
+            action: Run services
+            dockerComposeFile: docker-compose.yml
+            buildImages: true
+
+        - task: UsePythonVersion@0
+          inputs:
+            versionSpec: '3.8'
+
+        - script: |
+            python -m venv env

```

Comment:
Additional "set -eux" to bash tasks for logging and errors

Reply:
Thanks missed that. exiting with nonzero code is not failing ado?

Reply:
Done